### PR TITLE
Do not mutate Children inplace

### PR DESCRIPTION
**Pull request by philippjfr**
Fixes https://github.com/holoviz/panel/issues/7414

### CODE REVIEW

File: panel/viewable.py [hunk 1]
```diff
@@ -1149,10 +1149,16 @@ def __init__(
...
-            val[:] = [
-                v if isinstance(v, Viewable) else panel(v)
-                for v in val
-            ]
+            new = []
+            mutated = False
+            for v in val:
+                if not isinstance(v, Viewable):
+                    n = panel(v)
+                    mutated |= v is not n
```
Comment (maximlt) [inline panel/viewable.py:1157]: Out of curiosity, what kind of object that is not a `Viewable` would calling `panel(obj)` on return itself?

**Review by maximlt — APPROVED** [review]
LGTM.

Should `ChildDict._transform_value` also be adapted not to update in place the dict? (I don't think I've ever written `grid.objects = objects` but I don't use grids much).

File: panel/viewable.py (hunk 1 above)
Comment (philippjfr) [reply panel/viewable.py:1157]: Good point, it shouldn't return anything else. Will leave this just in case.

**Comment by philippjfr** [comment]
>Should ChildDict._transform_value also be adapted not to update in place the dict? (I don't think I've ever written grid.objects = objects but I don't use grids much).

Yes it should.

**Pull request merged**